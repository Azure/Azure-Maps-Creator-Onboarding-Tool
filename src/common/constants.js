export const PATHS = {
  INDEX: '/',
  VIEW_CONVERSIONS: '/view-conversions',
  CREATE_UPLOAD: '/create-upload',
  PROCESSING: '/processing',
  CREATE_GEOREFERENCE: '/create-georeference',
  REVIEW_CREATE: '/review-create',
  CONVERSION: '/review-create/conversion',
  IMDF_CONVERSION: '/review-create/imdf-conversion',
  PAST_CONVERSION: '/past-conversion',
  CONVERSIONS: '/conversions',
  LAYERS: '/layers',
  LEVELS: '/levels',
  INVALID_PATH: '/*',
};

export const ROUTE_NAME_BY_PATH = {
  [PATHS.INDEX]: 'home',
  [PATHS.CREATE_UPLOAD]: 'Create',
  [PATHS.VIEW_CONVERSIONS]: 'View',
  [PATHS.PROCESSING]: 'processing',
  [PATHS.CREATE_GEOREFERENCE]: 'prepare.drawing.package',
  [PATHS.LAYERS]: 'prepare.drawing.package',
  [PATHS.LEVELS]: 'prepare.drawing.package',
  [PATHS.INVALID_PATH]: 'redirect',
  [PATHS.REVIEW_CREATE]: 'prepare.drawing.package',
  [PATHS.CONVERSION]: 'create.indoor.map',
  [PATHS.IMDF_CONVERSION]: 'convert.indoor.map',
  [PATHS.CONVERSIONS]: 'All conversions',
};

export const PLACES_ROUTE_NAME_BY_PATH = {
  [PATHS.CREATE_GEOREFERENCE]: 'prepare.building.configuration',
  [PATHS.LAYERS]: 'prepare.building.configuration',
  [PATHS.LEVELS]: 'prepare.building.configuration',
  [PATHS.REVIEW_CREATE]: 'prepare.building.configuration',
};

export const TRUNCATE_FRACTION_DIGITS = 8;

export const HTTP_STATUS_CODE = {
  ACCEPTED: 202,
  OK: 200,
  BAD_REQUEST: 400,
  UNAUTHORIZED: 401,
  FORBIDDEN: 403,
  TOO_MANY_REQUESTS: 429,
};

export const isStagingEnv = process.env.REACT_APP_STAGING_ENV === 'true';

export const PLACES_PREVIEW = {
  DEFAULT_LANGUAGE: 'en',
  VERSION: 'places-1.0',
  DESCRIPTION: 'Converted by AzMapsCreatorOnboardingTool',
  STORAGE_RETENTION: 'day',
  SEARCH_PARAMETER: 'placespreview',
};
