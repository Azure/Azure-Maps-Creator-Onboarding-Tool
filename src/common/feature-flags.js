const flags = {
  onboardingEnabled: false,
};

export default flags;