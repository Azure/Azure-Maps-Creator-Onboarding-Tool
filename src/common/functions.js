/**
 * Given a single path name, returns a list of all path names from root to the given path name.
 * Ex. Given '/create/hello', returns ['/', '/create', '/create/hello']
 * @param {String} pathName the path name of the URL
 * @returns {String[]} an array of path names
 */
export const getSplitPaths = (pathName) => {
  if (typeof pathName !== 'string' || pathName === '') { return []; }
  if (pathName === '/') { return ['/']; }

  const pathNameLowerCase = pathName.toLowerCase();
  const splits = pathNameLowerCase.split('/');
  const splitPaths = [];
  for (let i = splits.length - 1; i > 0; i--) {
    let stringBuilder = '';
    for (let j = 1; j <= i; j++) {
      stringBuilder += '/' + splits[j];
    }
    splitPaths.push(stringBuilder);
  }
  splitPaths.push('/');
  splitPaths.reverse();

  return splitPaths;
};