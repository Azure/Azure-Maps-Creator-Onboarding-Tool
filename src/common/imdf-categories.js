export const imdfCategories = [
  'auditorium',
  'brick',
  'classroom',
  'column',
  'concrete',
  'conferenceroom',
  'drywall',
  'elevator',
  'escalator',
  'fieldofplay',
  'firstaid',
  'fitnessroom',
  'foodservice',
  'footbridge',
  'glass',
  'huddleroom',
  'kitchen',
  'laboratory',
  'library',
  'lobby',
  'lounge',
  'mailroom',
  'mothersroom',
  'movietheater',
  'movingwalkway',
  'nonpublic',
  'office',
  'opentobelow',
  'parking',
  'phoneroom',
  'platform',
  'privatelounge',
  'ramp',
  'recreation',
  'restroom',
  'restroom.family',
  'restroom.female',
  'restroom.female.wheelchair',
  'restroom.male',
  'restroom.male.wheelchair',
  'restroom.transgender',
  'restroom.transgender.wheelchair',
  'restroom.unisex',
  'restroom.unisex.wheelchair',
  'restroom.wheelchair',
  'road',
  'room',
  'serverroom',
  'shower',
  'smokingarea',
  'stairs',
  'steps',
  'storage',
  'structure',
  'terrace',
  'theater',
  'unenclosedarea',
  'unspecified',
  'vegetation',
  'waitingroom',
  'walkway',
  'walkway.island',
  'wood',
];
