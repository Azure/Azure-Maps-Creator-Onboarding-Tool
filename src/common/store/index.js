export { conversionSteps, useConversionStore } from './conversion.store';
export { useDissolvedExterior, useGeometryStore } from './geometry.store';
export { useLayersStore } from './layers.store';
export { useLevelsStore } from './levels.store';
export {
  progressBarStepsByKey,
  useCompletedSteps,
  useProgressBarSteps,
  useProgressBarStore,
} from './progress-bar-steps';
export { resetStores } from './reset';
export { LRO_STATUS, useResponseStore } from './response.store';
export { useReviewManifestJson, useReviewManifestStore } from './review-manifest.store';
export { getDomain, useUserStore } from './user.store';
