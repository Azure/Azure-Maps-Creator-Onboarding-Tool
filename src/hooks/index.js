import useAlert from './useAlert';
import useCustomNavigate from './useCustomNavigate';
import useEventListener from './useEventListener';
import useFeatureFlags from './useFeatureFlags';

export { useAlert, useCustomNavigate, useEventListener, useFeatureFlags };
