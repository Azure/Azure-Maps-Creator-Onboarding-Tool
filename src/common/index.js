import { PATHS, ROUTE_NAME_BY_PATH } from './constants';

export { PATHS, ROUTE_NAME_BY_PATH };