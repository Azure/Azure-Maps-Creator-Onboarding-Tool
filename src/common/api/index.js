// The following API is designed to be used only by the onboarding tool and is not supported for any other use.
import { getEnvs } from 'common/functions';
import { useUserStore } from '../store/user.store';

export const uploadFile = (file) => {
  const { geography, subscriptionKey } = useUserStore.getState();
  const url = `${getEnvs()[geography].URL}/manifest?api-version=2.0&subscription-key=${subscriptionKey}`;

  return fetch(url, {
    method: 'POST',
    headers: {
      'Content-Type': 'application/zip',
    },
    body: file,
  });
};

export const fetchFromLocation = (location) => {
  const { subscriptionKey } = useUserStore.getState();
  return fetch(`${location}&subscription-key=${subscriptionKey}`, {
    method: 'GET',
  });
};

export const deleteFromLocation = (fetchUrl) => {
  const { subscriptionKey } = useUserStore.getState();
  fetch(`${fetchUrl}&subscription-key=${subscriptionKey}`, {
    method: 'DELETE',
  });
};

export const fetchAddress = (address) => {
  const { geography, subscriptionKey } = useUserStore.getState();
  return fetch(`${getEnvs()[geography].URL}/search/address/json?subscription-key=${subscriptionKey}&api-version=1.0&query=${address}&limit=1`)
    .then((res) => res.json());
};