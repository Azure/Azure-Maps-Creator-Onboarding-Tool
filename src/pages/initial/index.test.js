import '@testing-library/jest-dom';
import { fireEvent, render, screen } from '@testing-library/react';
import { PATHS } from 'common';
import * as hooks from 'hooks';
import React from 'react';
import InitialView from './index';

jest.mock('react-i18next', () => ({
  useTranslation: () => ({
    t: key => key,
  }),
}));

jest.mock('hooks', () => ({
  useCustomNavigate: jest.fn(),
}));

describe('InitialView Component', () => {
  const mockedNavigate = jest.fn();

  beforeEach(() => {
    mockedNavigate.mockClear();
    hooks.useCustomNavigate.mockImplementation(() => mockedNavigate);
  });

  it('matches the snapshot', () => {
    const { asFragment } = render(<InitialView />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders correctly', () => {
    render(<InitialView />);
    expect(screen.getByText('Azure Maps Creator')).toBeInTheDocument();
    expect(screen.getByTestId('upload-button')).toBeInTheDocument();
    expect(screen.getByTestId('view-button')).toBeInTheDocument();
    expect(screen.getByText('initial.description')).toBeInTheDocument();
  });

  it('navigates to create upload on create button click', () => {
    render(<InitialView />);
    fireEvent.click(screen.getByTestId('upload-button'));
    expect(mockedNavigate).toHaveBeenCalledWith(PATHS.CREATE_UPLOAD);
  });

  it('navigates to view conversions on view button click', () => {
    render(<InitialView />);
    fireEvent.click(screen.getByTestId('view-button'));
    expect(mockedNavigate).toHaveBeenCalledWith(PATHS.VIEW_CONVERSIONS);
  });
});
