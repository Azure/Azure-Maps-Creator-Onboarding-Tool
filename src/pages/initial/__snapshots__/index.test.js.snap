// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InitialView Component matches the snapshot 1`] = `
<DocumentFragment>
  .emotion-0 {
  max-width: 45.5rem;
  width: 100%;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  font-size: 0.813rem;
  margin-top: 1rem;
}

.emotion-1 {
  margin-top: 1.5rem;
  margin-bottom: 1.5rem;
}

.emotion-2 {
  margin: 0 0 1.5rem 0;
}

.emotion-3 {
  height: 1.5rem;
  line-height: 1.5rem;
  font-size: 0.813rem;
}

<div
    class="emotion-0"
  >
    <div
      style="text-align: center;"
    >
      <h1
        class="emotion-1"
      >
        Azure Maps Creator
      </h1>
      <p
        class="emotion-2"
      >
        initial.description
      </p>
    </div>
    <div
      style="display: flex; justify-content: space-between; margin-top: 4rem;"
    >
      <div>
        <img
          alt="card"
          src="create.png"
          style="border-radius: 4px; margin-bottom: 0.5rem;"
        />
        <h2>
          Create
        </h2>
        <div>
          <p>
            create.description
          </p>
          <button
            class="ms-Button ms-Button--primary emotion-3 root-109"
            data-is-focusable="true"
            data-testid="upload-button"
            style="margin-top: 2rem;"
            type="button"
          >
            <span
              class="ms-Button-flexContainer flexContainer-110"
              data-automationid="splitbuttonprimary"
            >
              <span
                class="ms-Button-textContainer textContainer-111"
              >
                <span
                  class="ms-Button-label label-113"
                  id="id__0"
                >
                  create
                </span>
              </span>
            </span>
          </button>
        </div>
      </div>
      <div>
        <img
          alt="card"
          src="view.png"
          style="border-radius: 4px; margin-bottom: 0.5rem;"
        />
        <h2>
          View
        </h2>
        <div>
          <p>
            view.existing.conversions
          </p>
          <button
            class="ms-Button ms-Button--primary emotion-3 root-109"
            data-is-focusable="true"
            data-testid="view-button"
            style="margin-top: 2rem;"
            type="button"
          >
            <span
              class="ms-Button-flexContainer flexContainer-110"
              data-automationid="splitbuttonprimary"
            >
              <span
                class="ms-Button-textContainer textContainer-111"
              >
                <span
                  class="ms-Button-label label-113"
                  id="id__3"
                >
                  view
                </span>
              </span>
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
